### ISSUE TITLE
Training history not being logged when using ultralytics library by cloning github repository

### ISSUE THREAD

**Issue by farhan-hafizh** [ultralytics/ultralytics#18520]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I'm training YOLOv10x on Kaggle env. Instead of doing `!pip install ultralytics` i forked the original repository then i `!pip install git+https://github.com/...` because later i want to modified something.

The installation is successfull but when i try to train the model like this :
`train_result = yolo_model.train(
    data="/kaggle/working/dataset.yaml",
    epochs=60,
    optimizer="AdamW",
    batch=16,
    lr0=0.002,
    lrf=0.001,
    patience=10,
    momentum=0.94,
    weight_decay=0.0005, 
    warmup_epochs=7,     
    warmup_momentum=0.8, 
    box=0.15,
    cls=2.4,
    imgsz=640,
    cache="disk",
    plots=True,
    save=True,
    augment=False,
    device=[0,1]
)`

Only this is being logged:
`Ultralytics 8.3.57 🚀 Python-3.10.12 torch-2.4.1+cu121 CUDA:0 (Tesla T4, 15095MiB)
                                                       CUDA:1 (Tesla T4, 15095MiB)
engine/trainer: task=detect, mode=train, model=yolov10x.yaml, data=/kaggle/working/dataset.yaml, epochs=60, time=None, patience=10, batch=16, imgsz=640, save=True, save_period=-1, cache=disk, device=[0, 1], workers=8, project=None, name=train, exist_ok=False, pretrained=True, optimizer=AdamW, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=True, opset=None, workspace=None, nms=False, lr0=0.002, lrf=0.001, momentum=0.94, weight_decay=0.0005, warmup_epochs=7, warmup_momentum=0.8, warmup_bias_lr=0.1, box=0.15, cls=2.4, dfl=1.5, pose=12.0, kobj=1.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, bgr=0.0, mosaic=1.0, mixup=0.0, copy_paste=0.0, copy_paste_mode=flip, auto_augment=randaugment, erasing=0.4, crop_fraction=1.0, cfg=None, tracker=botsort.yaml, save_dir=runs/detect/train
Downloading https://ultralytics.com/assets/Arial.ttf to '/root/.config/Ultralytics/Arial.ttf'...
100%|██████████| 755k/755k [00:00<00:00, 18.7MB/s]
Overriding model.yaml nc=80 with nc=4

                   from  n    params  module                                       arguments                     
  0                  -1  1      2320  ultralytics.nn.modules.conv.Conv             [3, 80, 3, 2]                 
  1                  -1  1    115520  ultralytics.nn.modules.conv.Conv             [80, 160, 3, 2]               
  2                  -1  3    436800  ultralytics.nn.modules.block.C2f             [160, 160, 3, True]           
  3                  -1  1    461440  ultralytics.nn.modules.conv.Conv             [160, 320, 3, 2]              
  4                  -1  6   3281920  ultralytics.nn.modules.block.C2f             [320, 320, 6, True]           
  5                  -1  1    213120  ultralytics.nn.modules.block.SCDown          [320, 640, 3, 2]              
  6                  -1  6   4604160  ultralytics.nn.modules.block.C2fCIB          [640, 640, 6, True]           
  7                  -1  1    417920  ultralytics.nn.modules.block.SCDown          [640, 640, 3, 2]              
  8                  -1  3   2712960  ultralytics.nn.modules.block.C2fCIB          [640, 640, 3, True]           
  9                  -1  1   1025920  ultralytics.nn.modules.block.SPPF            [640, 640, 5]                 
 10                  -1  1   1545920  ultralytics.nn.modules.block.PSA             [640, 640]                    
 11                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 12             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 13                  -1  3   3122560  ultralytics.nn.modules.block.C2fCIB          [1280, 640, 3, True]          
 14                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 15             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 16                  -1  3   1948800  ultralytics.nn.modules.block.C2f             [960, 320, 3]                 
 17                  -1  1    922240  ultralytics.nn.modules.conv.Conv             [320, 320, 3, 2]              
 18            [-1, 13]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 19                  -1  3   2917760  ultralytics.nn.modules.block.C2fCIB          [960, 640, 3, True]           
 20                  -1  1    417920  ultralytics.nn.modules.block.SCDown          [640, 640, 3, 2]              
 21            [-1, 10]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 22                  -1  3   3122560  ultralytics.nn.modules.block.C2fCIB          [1280, 640, 3, True]          
 23        [16, 19, 22]  1   4392744  ultralytics.nn.modules.head.v10Detect        [4, [320, 640, 640]]          
YOLOv10x summary: 688 layers, 31,662,584 parameters, 31,662,568 gradients, 171.0 GFLOPs

DDP: debug command /usr/bin/python3 -m torch.distributed.run --nproc_per_node 2 --master_port 34925 /root/.config/Ultralytics/DDP/_temp_5bhqnr98140542120072272.py`

Is it a bug or i did something wrong?

### Additional

Ultralytics 8.3.57 🚀 Python-3.10.12 torch-2.4.1+cu121 CUDA:0 (Tesla T4, 15095MiB)
Setup complete ✅ (4 CPUs, 31.4 GB RAM, 6037.7/8062.4 GB disk)

OS                  Linux-6.6.56+-x86_64-with-glibc2.35
Environment         Colab
Python              3.10.12
Install             pip
RAM                 31.35 GB
Disk                6037.7/8062.4 GB
CPU                 Intel Xeon 2.00GHz
CPU count           4
GPU                 Tesla T4, 15095MiB
GPU count           2
CUDA                12.1

numpy               ✅ 1.26.4>=1.23.0
numpy               ✅ 1.26.4<2.0.0; sys_platform == "darwin"
matplotlib          ✅ 3.7.1>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 10.4.0>=7.1.2
pyyaml              ✅ 6.0.2>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.13.1>=1.4.1
torch               ✅ 2.4.1+cu121>=1.8.0
torch               ✅ 2.4.1+cu121!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision         ✅ 0.19.1+cu121>=0.9.0
tqdm                ✅ 4.66.5>=4.64.0
psutil              ✅ 5.9.5
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.1.4>=1.1.4
seaborn             ✅ 0.12.2>=0.11.0
ultralytics-thop    ✅ 2.0.13>=2.0.0
{'OS': 'Linux-6.6.56+-x86_64-with-glibc2.35',
 'Environment': 'Colab',
 'Python': '3.10.12',
 'Install': 'pip',
 'RAM': '31.35 GB',
 'Disk': '6037.7/8062.4 GB',
 'CPU': 'Intel Xeon 2.00GHz',
 'CPU count': 4,
 'GPU': 'Tesla T4, 15095MiB',
 'GPU count': 2,
 'CUDA': '12.1',
 'Package Info': {'numpy': '✅ 1.26.4<2.0.0; sys_platform == "darwin"',
  'matplotlib': '✅ 3.7.1>=3.3.0',
  'opencv-python': '✅ 4.10.0.84>=4.6.0',
  'pillow': '✅ 10.4.0>=7.1.2',
  'pyyaml': '✅ 6.0.2>=5.3.1',
  'requests': '✅ 2.32.3>=2.23.0',
  'scipy': '✅ 1.13.1>=1.4.1',
  'torch': '✅ 2.4.1+cu121!=2.4.0,>=1.8.0; sys_platform == "win32"',
  'torchvision': '✅ 0.19.1+cu121>=0.9.0',
  'tqdm': '✅ 4.66.5>=4.64.0',
  'psutil': '✅ 5.9.5',
  'py-cpuinfo': '✅ 9.0.0',
  'pandas': '✅ 2.1.4>=1.1.4',
  'seaborn': '✅ 0.12.2>=0.11.0',
  'ultralytics-thop': '✅ 2.0.13>=2.0.0'}}

**Comment by UltralyticsAssistant** [comment]
👋 Hello @farhan-hafizh, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) (MRE) to help us debug it. From your description, it seems the logging issue may be related to the DDP (Distributed Data Parallel) or custom repository installation. Please share additional relevant details such as:

1. The specific changes you made to the repository, if any.
2. A stripped-down version of your training script that shows the problem with as few lines of code as possible.
3. Logs after reducing the GPUs (e.g., setting `device=0` instead of `[0,1]`) to narrow down the cause.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response, but an Ultralytics engineer will assist you further shortly. Thank you for your patience and for contributing to improving Ultralytics! 😊

**Comment by farhan-hafizh** [comment]
You can reproduce this without forking

**Comment by farhan-hafizh** [comment]
I currently found out that the behaviour only happen in kaggle, at least on google colab the history is being logged. Also in kaggle if i did `!pip install ultralytics` instead of using git, the training history is being logged.

**Comment by Y-T-G** [comment]
Someone else also reported multi-GPU training with Kaggle doesn't show logs

**Comment by farhan-hafizh** [comment]
i dont know but i think it only happen if the package is installed from git, because if i do `!pip install ultralytics` the logs was shown. even on wandb only system log that exist.

**Comment by glenn-jocher** [comment]
Thank you for the clarification. This issue may stem from differences in how logging is configured between the PyPI and GitHub installations. If you're using the GitHub version, ensure all dependencies are properly installed and logging tools like `wandb` are correctly initialized. For consistent logging behavior, we recommend using the PyPI version unless modifications are required.
